accepted/0000-argument-option-parsing.md

```diff
@@ -0,0 +1,42 @@
+# Option and Argument Parsing for npm
+
+## Summary
+
+Currently, option, and argument handling in npm is not using any kind of standard library (either internal or a third party lib). This means that invalid arguments or flags are not detected.
+
+## Motivation
+
+Originally proposed in [this issue](https://github.com/npm/npm/issues/20856).
+
+Because there's no error thrown when invalid command-line flags are passed, you can pass in invalid flags and not realize they're invalid for a particular context. e.g., if you type: `npm audit --format=json` instead of `npm audit --json` you get the normal output of `npm audit` without any arguments, vs. an error that you passed an invalid flag. This should probably be fixed regardless of the outcome of this discussion / RFC.
+
+Additionally, if the command line flags and args are built and documented using a standard library, there's less chance for drift between usage / docs and reality as new subcommands, flags, and arguments are added. After the initial work, this should also make implementing new options easier.
+
+I'm guessing it could also make keeping completion up to date easier.
+
+## Detailed Explanation
+
+This would pull in a third party lib, and / or have a new library (in `utils/`, probably) that would be used to declare subcommands, arguments, and options in a standardized way. Ideally, it should be self-documenting (so that usage text, auto-completion, help pages, etc. could be as auto-generated as possible).
+
+Then, npm itself, as well as its subcommands, could be cut over to use the new lib, either all at once or, more likely, one at a time, and eventually, older utilities could be deprecated or removed.
+
+## Rationale and Alternatives
+
+
+## Implementation
+
```

Comment:
at least some guesswork here might be good. tl;dr this would have to be done by `lib/config/core.js` or possible `lib/config/defaults.js`.

Reply:
Thanks - I put that in verbatim, HTH. I haven't worked with npm yet, and so I'm guessing others could probably make a much more informed decision about _how_ this should be implemented; happy to put in more information in response to feedback, with or without attribution.